<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Editor</class>
 <widget class="QMainWindow" name="Editor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>578</width>
    <height>362</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Yukkuri Editor</string>
  </property>
  <property name="animated">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <widget class="QWidget" name="widget" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="minimumSize">
          <size>
           <width>150</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>230</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="title">
          <string>Files</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QListWidget" name="FilesList">
            <property name="editTriggers">
             <set>QAbstractItemView::DoubleClicked</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="NewConfigButton">
            <property name="text">
             <string>New config file</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QListWidget" name="ItemsList"/>
          </item>
          <item>
           <widget class="QPushButton" name="NewEntryButton">
            <property name="text">
             <string>New entry</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QTabWidget" name="MainTabs">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="YSpritesTab" name="SpriteTab">
            <attribute name="title">
             <string>Sprites</string>
            </attribute>
           </widget>
           <widget class="YEntityTab" name="EntityTab">
            <attribute name="title">
             <string>Entity</string>
            </attribute>
           </widget>
           <widget class="YWidgetsTab" name="WidgetsTab">
            <attribute name="title">
             <string>Widgets</string>
            </attribute>
           </widget>
           <widget class="YTilesTab" name="MapTilesTab">
            <attribute name="title">
             <string>Tiles</string>
            </attribute>
           </widget>
           <widget class="YMapTab" name="MapTab">
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <attribute name="title">
             <string>Map</string>
            </attribute>
           </widget>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="folder">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Source files directory path:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="FolderPlace">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDialogButtonBox" name="OpenFolderDBox">
        <property name="maximumSize">
         <size>
          <width>75</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="standardButtons">
         <set>QDialogButtonBox::Open</set>
        </property>
        <property name="centerButtons">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QDialogButtonBox" name="exitbox">
        <property name="standardButtons">
         <set>QDialogButtonBox::Close|QDialogButtonBox::RestoreDefaults|QDialogButtonBox::Save</set>
        </property>
        <property name="centerButtons">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>578</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuEditor">
    <property name="title">
     <string>Editor</string>
    </property>
    <addaction name="actionGeneralConfig"/>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuEditor"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionGeneralConfig">
   <property name="text">
    <string>General config</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>YWidgetsTab</class>
   <extends>QWidget</extends>
   <header>widgets</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>YEntityTab</class>
   <extends>QWidget</extends>
   <header>widgets</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>YTilesTab</class>
   <extends>QWidget</extends>
   <header>widgets</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>YMapTab</class>
   <extends>QWidget</extends>
   <header>widgets</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>YSpritesTab</class>
   <extends>QWidget</extends>
   <header>widgets</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>FolderPlace</tabstop>
  <tabstop>exitbox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>OpenFolderDBox</sender>
   <signal>clicked(QAbstractButton*)</signal>
   <receiver>Editor</receiver>
   <slot>openFolder()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>567</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>598</x>
     <y>544</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NewConfigButton</sender>
   <signal>clicked()</signal>
   <receiver>Editor</receiver>
   <slot>createConfigFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>335</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NewEntryButton</sender>
   <signal>clicked()</signal>
   <receiver>Editor</receiver>
   <slot>createConfigRecord()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>66</x>
     <y>249</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>544</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exitbox</sender>
   <signal>accepted()</signal>
   <receiver>Editor</receiver>
   <slot>saveFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>567</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>598</x>
     <y>544</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>Editor</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>410</x>
     <y>377</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainTabs</sender>
   <signal>currentChanged(int)</signal>
   <receiver>Editor</receiver>
   <slot>reloadFiles(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>487</x>
     <y>19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>FilesList</sender>
   <signal>itemActivated(QListWidgetItem*)</signal>
   <receiver>Editor</receiver>
   <slot>reloadElements(QListWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>84</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ItemsList</sender>
   <signal>itemActivated(QListWidgetItem*)</signal>
   <receiver>Editor</receiver>
   <slot>reloadContent(QListWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>442</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exitbox</sender>
   <signal>rejected()</signal>
   <receiver>Editor</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>420</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>410</x>
     <y>377</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openFolder()</slot>
  <slot>createConfigFile()</slot>
  <slot>createConfigRecord()</slot>
  <slot>saveFile()</slot>
  <slot>reloadFiles(int)</slot>
  <slot>reloadElements(QListWidgetItem*)</slot>
  <slot>reloadContent(QListWidgetItem*)</slot>
 </slots>
</ui>
